packages/core/src/provider/ComposioProvider.ts

```diff
@@ -11,9 +11,11 @@ import { BaseNonAgenticProvider } from './BaseProvider';
 interface CustomTool {
   name: string;
 }
-export class ComposioProvider extends BaseNonAgenticProvider<Array<CustomTool>, CustomTool> {
+export class ComposioProvider extends BaseNonAgenticProvider<Array<CustomTool>, CustomTool, McpServerGetResponse> {
   readonly name = 'ComposioProvider';
 
+  readonly mcp = new BaseMcpProvider<McpServerGetResponse>();
```

Comment:
If the type of BaseMcpProvider can be hardcoded, we do not need type generics here. This should be simply `new BaseMcpProvider()`

Reply:
makes sense, miss on my end.